setuptools/command/bdist_egg.py

```diff
@@ -238,13 +239,26 @@ def run(self):
 
     def zap_pyfiles(self):
         log.info("Removing .py files from temporary directory")
+        py3 = sys.version_info >= (3, 0)
+        re_pycache_file = re.compile('(.+)\.cpython-3\d(.pyc)$')
         for base, dirs, files in walk_egg(self.bdist_dir):
             for name in files:
                 if name.endswith('.py'):
                     path = os.path.join(base, name)
                     log.debug("Deleting %s", path)
                     os.unlink(path)
 
+                if py3 and base.endswith('__pycache__'):
+                    path_old = os.path.join(base, name)
+                    
+                    m = re_pycache_file.match(name)
+                    path_new = os.path.join(base, os.pardir, m.group(1) + m.group(2))
+                    log.info("Renaming Python 3 .pyc file from [%s] to [%s]" % (path_old, path_new))
+                    if os.path.exists(path_new):
```

Comment:
There's a race condition here. Better to simply attempt to remove it and suppress any failure Exception.